### PR TITLE
Merge 2.2 to Master

### PR DESCRIPTION
**Pull request by jkotalik**
#6761

Please confirm all changes are done correctly for area owners.

### CODE REVIEW

File: src/Servers/Kestrel/Core/test/Http1ConnectionTests.cs [hunk 1]
```diff
@@ -724,6 +728,20 @@ public async Task RequestAbortedTokenIsResetBeforeLastWriteWithChunkedEncoding()
...
+            var originalRegistration = originalToken.Register(() => { });
+
+            _http1Connection.Abort(new ConnectionAbortedException());
+
+            Assert.True(originalToken.WaitHandle.WaitOne(TestConstants.DefaultTimeout));
+            Assert.True(_http1Connection.RequestAborted.WaitHandle.WaitOne(TestConstants.DefaultTimeout));
+
+            Assert.Equal(originalToken, originalRegistration.Token);
+        }
+
```
Comment (halter73) [inline src/Servers/Kestrel/Core/test/Http1ConnectionTests.cs]: The Kestrel part LGTM. Thanks @jkotalik 🎉

**Review by JamesNK — APPROVED** [review]
Routing changes from @rynowak and I look good

**Review by Tratcher — APPROVED** [review]
For Auth Samples

File: src/Security/samples/PathSchemeSelection/PathSchemeSelection.csproj [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <IsTestAssetProject>true</IsTestAssetProject>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Reference Include="Microsoft.AspNetCore" />
+    <Reference Include="Microsoft.AspNetCore.Authentication.Cookies" />
+    <Reference Include="Microsoft.AspNetCore.Mvc" />
+    <Reference Include="Microsoft.AspNetCore.StaticFiles" />
+    <Reference Include="Microsoft.NET.Sdk.Razor" PrivateAssets="All" />
```
Comment (jkotalik) [inline src/Security/samples/PathSchemeSelection/PathSchemeSelection.csproj]: @natemcmaster these references were required in netcoreapp3.0. I remember you saying Microsoft.NET.Sdk.Web would bring it in.

Comment (natemcmaster) [reply src/Security/samples/PathSchemeSelection/PathSchemeSelection.csproj]: This is fine for now. We'll clean this up later. We need to cleanup a bunch of other projects in master, and we can do that in one sweep.

File: src/Framework/test/SharedFxTests.cs [hunk 3]
```diff
@@ -114,19 +114,33 @@ public void ItContainsValidDepsJson()
             Assert.NotNull(depsFile["compilationOptions"]);
             Assert.Empty(depsFile["compilationOptions"]);
             Assert.NotEmpty(depsFile["runtimes"][_expectedRid]);
-            Assert.All(depsFile["libraries"], item =>
```
Comment (natemcmaster) [inline src/Framework/test/SharedFxTests.cs]: Can you undo the changes to this test? The test in master is correct and should be left untouched. The changes I made in 2.x were to backport a portion of this test.

**Review by natemcmaster — APPROVED** [review]
LGTM.

**Review by halter73 — APPROVED** [review]

**Comment by jkotalik** [comment]
Kestrel InMemory.FunctionalTests/netcoreapp3.0 hung on the last test run.

**Comment by jkotalik** [comment]
Alright, I have hit ~5 runs of test failures:
https://github.com/aspnet/AspNetCore/pull/6772/checks?check_run_id=51058950
DATA_Sent_TooSlowlyDueToFlowControlOnSmallWrite_AbortsConnectionAfterGracePeriod
Microsoft.DotNet.Watcher.Tools.Tests.MsBuildFileSetFactoryTest.SingleTfm

I'm going to try running this one more time and if it fails again, someone with admin will have to merge it.

**Comment by jkotalik** [comment]
@ajaybhargavb make sure the test fixes you add to DefaultRazorProjectFileSystemTest.cs and RazorCompileIntegrationTest.cs get ported to https://github.com/aspnet/aspnetcore-tooling
